packages/widgets/src/index.js

```diff
@@ -18,11 +18,15 @@ export * from './utils';
  * Note that for the block to be useful, any scripts required by a widget must
  * be loaded into the page.
  *
+ * @param {Object} settings Block settings.
  * @see https://developer.wordpress.org/block-editor/how-to-guides/widgets/legacy-widget-block/
  */
-export function registerLegacyWidgetBlock() {
-	const { metadata, settings, name } = legacyWidget;
-	registerBlockType( { name, ...metadata }, settings );
+export function registerLegacyWidgetBlock( settings = {} ) {
```

Comment:
Maybe the argument could be just "supportsInserter", do we want to afford other modifications in the future?

Reply:
Or we can just allow overriding the  `supports` settings.

Reply:
Updated in [0a1c3df](https://github.com/WordPress/gutenberg/pull/45509/commits/0a1c3dff0059e2562aff497778a6c04278aa4651).